sdk/python/feast/constants.py

```diff
@@ -21,135 +21,188 @@ class AuthProvider(Enum):
+
+    #: Enable or disable control loop for Feast Job Service
+    JOB_SERVICE_ENABLE_CONTROL_LOOP: str = "False"
+
+    #: Default connection timeout to Feast Serving, Feast Core, and Feast Job Service (in seconds)
+    GRPC_CONNECTION_TIMEOUT: str = "10"
+
+    #: Default gRPC connection timeout when sending an ApplyFeatureSet command to Feast Core (in seconds)
+    GRPC_CONNECTION_TIMEOUT_APPLY: str = "600"
+
+    #: Default timeout when running batch ingestion
+    BATCH_INGESTION_PRODUCTION_TIMEOUT: str = "120"
+
+    #: Time to wait for historical feature requests before timing out.
+    BATCH_FEATURE_REQUEST_WAIT_TIME_SECONDS: str = "600"
+
+    #: Authentication Provider - Google OpenID/OAuth
+    AUTH_PROVIDER: str = "google"
+
+    #: Spark Job launcher
+    SPARK_LAUNCHER: str = "dataproc"  # standalone, dataproc, emr
+
+    #: Feast Spark Job ingestion jobs staging location
+    SPARK_STAGING_LOCATION: str = ""
+
+    #: Feast Spark Job ingestion jar file
+    SPARK_INGESTION_JAR: str = "https://storage.googleapis.com/feast-jobs/spark/ingestion/feast-ingestion-spark-develop.jar"
+
+    #: Spark resource manager master url
+    SPARK_STANDALONE_MASTER: str = "local[*]"
+
+    #: Directory where Spark is installed
+    SPARK_HOME: str = ""
+
+    #: Dataproc cluster to run Feast Spark Jobs in
+    DATAPROC_CLUSTER_NAME: str = ""
+
+    #: Project of Dataproc cluster
+    DATAPROC_PROJECT: str = ""
+
+    #: Region of Dataproc cluster
+    DATAPROC_REGION: str = ""
+
+    #: File format of historical retrieval features
+    HISTORICAL_FEATURE_OUTPUT_FORMAT: str = "parquet"
+
+    #: File location of historical retrieval features
+    HISTORICAL_FEATURE_OUTPUT_LOCATION: str = ""
+
+    #: Default Redis host
+    REDIS_HOST: str = "localhost"
+
+    #: Default Redis port
+    REDIS_PORT: str = "6379"
+
+    #: Enable or disable TLS/SSL to Redis
+    REDIS_SSL: str = "False"
+
+    #: Enable or disable StatsD
+    STATSD_ENABLED: str = "False"
+
+    #: Default StatsD port
+    STATSD_HOST: str = ""
+
+    #: Default StatsD port
+    STATSD_PORT: str = ""
+
+    #: IngestionJob DeadLetter Destination
+    DEADLETTER_PATH: str = ""
+
+    #: ProtoRegistry Address (currently only Stencil Server is supported as registry)
+    #: https://github.com/gojekfarm/stencil
+    STENCIL_URL: str = ""
+
+    #: EMR cluster to run Feast Spark Jobs in
+    EMR_CLUSTER_ID: str = ""
+
+    #: Region of EMR cluster
+    EMR_REGION: str = ""
+
+    #: Template path of EMR cluster
+    EMR_CLUSTER_TEMPLATE_PATH: str = ""
+
+    #: Log path of EMR cluster
+    EMR_LOG_LOCATION: str = ""
+
+    #: Oauth grant type
+    OAUTH_GRANT_TYPE: str = ""
+
+    #: Oauth client ID
+    OAUTH_CLIENT_ID: str = ""
+
+    #: Oauth client secret
+    OAUTH_CLIENT_SECRET: str = ""
+
+    #: Oauth intended recipients
+    OAUTH_AUDIENCE: str = ""
+
+    #: Oauth token request url
+    OAUTH_TOKEN_REQUEST_URL: str = ""
+
+    MAX_WAIT_INTERVAL: str = "60"
```

Comment:
Excluded docs for this on purpose since it's only used internally.

Reply:
What do you mean by `internally`?

Reply:
It's just strange how we allow users to set it but we dont document it. I think it should be documented.

Reply:
Shifted out of configurable options.